### PR TITLE
Fix FbExceptionTests(Embedded).SQLSTATETest

### PR DESCRIPTION
**Pull request by zabulus**

### CODE REVIEW

File: NETProvider/src/FirebirdSql.Data.FirebirdClient/Common/IscException.cs [hunk 1]
```diff
@@ -187,7 +187,7 @@ private void BuildSqlState()
 		{
 			IscError error = this.Errors.Find(e => e.Type == IscCodes.isc_arg_sql_state);
 			// step #1, maybe we already have a SQLSTATE stuffed in the status vector
-			if (error != null)
+			if (error != null && !string.IsNullOrEmpty(error.StrParam))
```
Comment (cincuranet) [inline NETProvider/src/FirebirdSql.Data.FirebirdClient/Common/IscException.cs]: What's the idea behind it?

Comment (zabulus) [inline NETProvider/src/FirebirdSql.Data.FirebirdClient/Common/IscException.cs]: ```error.StrParam``` is null when test launched in embedded config

Comment (cincuranet) [inline NETProvider/src/FirebirdSql.Data.FirebirdClient/Common/IscException.cs]: Interesting. I'll check with engine guys. Just to make sure we're not loosing the content somewhere.

**Comment by zabulus** [comment]
Reimplemented fix. Found out missing parsing of sql state.

**Pull request merged**